auth/providers/azure/azure.go

```diff
@@ -120,23 +122,34 @@ type metadataJSON struct {
 func getMetadata(aadEndpoint, tenantID string) (*metadataJSON, error) {
 	metadataURL := aadEndpoint + tenantID + "/.well-known/openid-configuration"
 
-	response, err := http.Get(metadataURL)
+	request, err := http.NewRequest("GET", metadataURL, nil)
```

Comment:
If we are going to just set the timeout for the request, I wonder is that possible to just bind a context with 3s timeout to the request? The sender is just a normal http.Client, it should obey the timeout from request context

Reply:
+100. NewRequestWithContext is designed for per request customization. this would be my preference as well.